tests/src/rules/no-extraneous-dependencies.js

```diff
@@ -154,5 +158,29 @@ ruleTester.run('no-extraneous-dependencies', rule, {
         message: '\'lodash.isarray\' should be listed in the project\'s dependencies, not optionalDependencies.',
       }],
     }),
+    test({
+      code: 'import "not-a-dependency"',
+      options: [{packagePath: path.join(__dirname, '../../../package.json')}],
+      errors: [{
+        ruleId: 'no-extraneous-dependencies',
+        message: '\'not-a-dependency\' should be listed in the project\'s dependencies. Run \'npm i -S not-a-dependency\' to add it',
+      }],
+    }),
+    test({
+      code: 'import "not-a-dependency"',
+      options: [{packagePath: path.join(__dirname, '../../../doesnt-exist.json')}],
```

Comment:
This kind of highlights a weirdness of this rule: you can specify a file that is not named `package.json`, which I don't think makes sense. I'm wondering whether `packagePath` should not rather be `packageDir` and be the path to the folder containing package.json.
i.e. `[{packagePath: path.join(__dirname, '../../../package.json')}]` --> `[{packageDir: path.join(__dirname, '../../../')}]`

What do you think? :thinking: